### PR TITLE
Support [x; n] expressions in concat_bytes!

### PR DESCRIPTION
**Pull request by syvb**
Currently trying to use `concat_bytes!` with a repeating array value like `[42; 5]` results in an error:
```
error: expected a byte literal
 --> src/main.rs:3:27
  |
3 |     let x = concat_bytes!([3; 4]);
  |                           ^^^^^^
  |
  = note: only byte literals (like `b"foo"`, `b's'`, and `[3, 4, 5]`) can be passed to `concat_bytes!()`
```

This makes it so repeating array syntax can be used the same way normal arrays can be. The RFC doesn't explicitly mention repeat expressions, but it seems reasonable to allow them as well, since normal arrays are allowed. 

It is possible to make the compiler get stuck compiling forever with `concat_bytes!([3; 999999999])`, but I don't think that's much of an issue since you can do that already with `const X: [u8; 999999999] = [3; 999999999];`.

Contributes to #87555.

### CODE REVIEW

**Comment by bjorn3** [comment]
What is the difference between `[1; 3]` and `concat_bytes!([1; 3])`?

File: compiler/rustc_builtin_macros/src/concat_bytes.rs [hunk 1]
```diff
@@ -88,48 +134,27 @@ pub fn expand_concat_bytes(
...
+                {
+                    if let Some(elem) =
+                        handle_array_element(cx, &mut has_errors, &mut missing_literals, expr)
+                    {
+                        for _ in 0..count_val {
+                            accumulator.push(elem);
                         }
                     }
+                } else {
+                    cx.span_err(count.value.span, "repeat count is not a number");
```
Comment (nagisa) [inline compiler/rustc_builtin_macros/src/concat_bytes.rs:157]: nit: maybe say “positive number”? `-2` is still a number.

**Comment by nagisa** [comment]
This seems reasonable to me.

**Comment by dtolnay** [comment]
FYI @rust-lang/libs-api. (`concat_bytes!` is still unstable)

**Comment by Smittyvb** [comment]
It looks like there was a spurious network error in CI:
```none
curl: (6) Could not resolve host: ci-mirrors.rust-lang.org
```

**Comment by matthiaskrgr** [comment]
lots of logs are missing :/
@bors retry rollup=iffy

**Pull request merged**